### PR TITLE
tools: webd - remember elektrad sessions

### PR DESCRIPTION
**Pull request by raphi011**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you for working on this! I assume this means rewriting webd in Go is not feasible?

**Comment by raphi011** [comment]
> Thank you for working on this! I assume this means rewriting webd in Go is not feasible?

Sure it's feasible, but the other issues have a higher priority right now :)

**Comment by raphi011** [comment]
@markus2330 @mpranj any idea why this build is failing? https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-3400/6/pipeline

**Comment by mpranj** [comment]
Yes, please rebase from master. There were some recent changes which fix this.

**Comment by raphi011** [comment]
This PR can be reviewed and merged if everything looks fine

File: src/tools/webd/src/routes/utils.js [hunk 1]
```diff
@@ -31,6 +31,35 @@ export const errorResponse = (res, err) => {
     .json({ error: errObj });
 };
 
+const sessionRegex = /session=([a-zA-Z0-9-]*)/;
+
+export const getSessionID = (instanceId, cookie) => {
+  instanceId = "asd"; // remove me
```
Comment (markus2330) [inline src/tools/webd/src/routes/utils.js:37]: remove me?

Comment (mpranj) [inline src/tools/webd/src/routes/utils.js:37]: Intentionally hardcoded?

File: src/tools/webd/src/connector.js [hunk 2]
```diff
@@ -16,14 +16,24 @@ const encodePath = path =>
 
 const version = host => fetch(`${host}/version`).then(res => res.json());
 
-const getRoot = (host, query = "") =>
-  fetch(`${host}/kdb${query}`).then(res => res.json());
+const getRoot = (host, sessionId, query = "") =>
```
Comment (markus2330) [inline src/tools/webd/src/connector.js:19]: Did this change the API?

File: src/tools/webd/src/routes/utils.js (hunk 1 above)
Comment (raphi011) [reply src/tools/webd/src/routes/utils.js:37]: ah, no this was only debug code .. good catch

File: src/tools/webd/src/connector.js (hunk 2 above)
Comment (raphi011) [reply src/tools/webd/src/connector.js:19]: no, we have to pass back the same session id per user to elektrad to allow reusage of a kdb handle

**Comment by markus2330** [comment]
As discussed via phone: this PR remembers the sessions in webd so that elektrad can work correctly.

**Pull request merged**